pkg/kapp/preflight/registry_test.go

```diff
@@ -102,7 +119,111 @@ func TestRegistryRun(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			err := tc.registry.Run(nil, nil)
-			require.Equal(t, tc.shouldErr, err != nil)
+			require.Equalf(t, tc.shouldErr, err != nil, "Unexpected error: %v", err)
+		})
+	}
+}
+
+func TestRegistryConfig(t *testing.T) {
+	testCases := []struct {
+		name       string
+		registry   *Registry
+		configYaml string
```

Comment:
Why should we use a YAML string here instead of directly using the `config.Conf` type?

My argument for using the `config.Conf` type over a YAML string is that there is less chance for errors to occur in the parsing of the YAML. Ideally we limit the potential for unrelated errors to occur as much as is reasonable for the piece of code we are testing. In this case, the `Registry.SetConfig()` method accepts a `config.Conf` as a parameter and building one of those directly in the unit test means we don't run the risk of potential changes in how the `config.Conf` is parsed from a YAML file breaking this unit test.

Reply:
The contents of `config.Conf` can't are private:
```
type Conf struct {
	configs []Config
}

Reply:
Gotcha. In that case, does `Registry.SetConfig` really need to take in the entire `Conf` type?

Reply:
You reference `PreflightConfigs()` below which does work off of a `config.Conf`, so that could work, but I prefer to work off the yaml config.